### ISSUE TITLE
Async AuthorisationStrategy?

### ISSUE THREAD

**Issue by avimar** [ForbesLindesay/connect-roles#3]
I wasn't sure where else to ask this.
How do I do an async auth strategy?

e.g. I have a role of editing their own resources -- so I match the credentials to the resource edit request.

But this new case is I want to allow the owner of many sub-accounts to edit the resource. So in that case, I need to do a query to find the owner of that account. (But only if other role-checks fail)

(or maybe I can do this a different way...)

So, is there some way to do async in the AuthorisationStrategy? there's no next(); to not call like in connect middleware...

Thanks!

**Comment by JorgeLGonzalez** [comment]
Any new thoughts on the promise-based solution for async authorisation strategy?

**Comment by dskrvk** [comment]
Can this be closed now?

**Comment by mikeatlas** [comment]
Works for me too. 

Note that it needs to only return Promises which either `resolve(true)` or `resolve(false)`, but not to use `reject(...)`:
```
permission.use('get stuff', req => (
  new Promise((resolve) => {
    if (userCanDoStuff(req)) { return resolve(true); }
    return resolve(false);
  })));
``

**Comment by mvolkmann** [comment]
Yes, works great for me also! Can the docs be changed to no longer say that the "async" option is experimental? My use case is to query a database to get the roles of the user which is why I needed the callback function of the "use" method to return a Promise. I would guess this is a common scenario. It would be great to show one example of an async function being passed to the "use" method in the docs.

**Comment by ForbesLindesay** [comment]
@mvolkmann Yes, if someone adds a test case to cover it.
